onnxruntime/core/optimizer/propagate_cast_ops.cc

```diff
@@ -655,13 +688,14 @@ static void SearchDownstream(Graph& graph, NodeArg* node_arg,
       }
     }
   }
-  if (graph.IsOutput(node_arg) && require_cast.find(node_arg) == require_cast.end()) {
-    require_cast.insert(std::make_pair(node_arg, std::vector<Node*>()));
+
+  if (graph.IsOutput(node_arg)) {
```

Comment:
is it intentional to remove the `!Contains(required_cast, node_arg)` here?

Reply:
Yes, avoids double lookup.